File: backend/src/main/java/com/ll/nbe342team8/domain/member/member/repository/MemberRepository.java
package com.ll.nbe342team8.domain.member.member.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.ll.nbe342team8.domain.member.member.entity.Member;

public interface MemberRepository extends JpaRepository<Member, Long> {
	Optional<Member> findByEmail(String email);

	Optional<Member> findByoAuthId(String oAuthId);

	// 관리자 계정 조회 (이메일ID, 회원 유형 확인)
	Optional<Member> findByEmailAndMemberType(String email, Member.MemberType memberType);

	// 관리자 계정 개수 확인
	long countByMemberType(Member.MemberType memberType);
}

File: backend/src/main/java/com/ll/nbe342team8/domain/member/member/repository/MemberRepository.kt
package com.ll.nbe342team8.domain.member.member.repository

import com.ll.nbe342team8.domain.member.member.entity.Member
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface MemberRepository : JpaRepository<Member, Long> {
    fun findByEmail(email: String): Optional<Member>

    fun findByoAuthId(oAuthId: String): Optional<Member>

    // 관리자 계정 조회 (이메일ID, 회원 유형 확인)
    fun findByEmailAndMemberType(email: String, memberType: Member.MemberType): Optional<Member>

    // 관리자 계정 개수 확인
    fun countByMemberType(memberType: Member.MemberType): Long
}
